pkg/cmd/task/start.go

```diff
@@ -277,28 +300,59 @@ func startTask(opt startOptions, args []string) error {
 			Namespace: opt.cliparams.Namespace(),
 		},
 	}
-
 	var tname string
-	if len(args) > 0 {
+
+	switch {
```

Comment:
here is where i think we can generalize the logic for a task and a pipeline? @vdemeester